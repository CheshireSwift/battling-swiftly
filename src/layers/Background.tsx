import * as React from 'react'

import { layerStyle } from '../helpers/styleSnippets'

export const Background = ({
  url,
  onLoadDimensions,
}: {
  url: string
  onLoadDimensions: (dimensions: { height: number; width: number }) => void
}) => (
  <img
    style={{ ...layerStyle, height: undefined, width: undefined }}
    src={url}
    onLoad={e => {
      const img = e.target as HTMLImageElement
      onLoadDimensions({ height: img.height, width: img.width })
    }}
  />
)

export default Background
