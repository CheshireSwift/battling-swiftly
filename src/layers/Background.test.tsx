import * as React from 'react'
import { mount, render } from 'enzyme'

import Background from './Background'

describe('the background layer', () => {
  it('matches the snapshot', () => {
    expect(
      render(
        <Background
          url="http://example.com/example.png"
          onLoadDimensions={jest.fn()}
        />,
      ),
    ).toMatchSnapshot()
  })

  it('renders an image with the given URL', () => {
    const url = 'http://example.com/example.png'
    const bg = render(<Background url={url} onLoadDimensions={jest.fn()} />)
    expect(bg.attr('src')).toEqual(url)
  })

  it('reports the dimensions of the loaded image', () => {
    const url = 'http://example.com/example.png'
    const dimensions = { width: 1, height: 2 }
    const reportDimensions = jest.fn()
    const bg = mount(
      <Background url={url} onLoadDimensions={reportDimensions} />,
    )
    bg.simulate('load', { target: { ...dimensions, foo: 'bar' } })
    expect(reportDimensions).toHaveBeenCalledWith(dimensions)
  })
})
