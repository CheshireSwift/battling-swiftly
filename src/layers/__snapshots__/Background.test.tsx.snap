// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the background layer matches the snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "src": "http://example.com/example.png",
      "style": "position:absolute;top:0;bottom:0",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "src": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "src": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
